pkg/crlprovider/cacheprovider/cacheprovider.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package crl
```

Comment:
wonder if we just use the cache provided by notation-go library? https://github.com/notaryproject/notation-go/blob/main/verifier/crl/crl.go

Seems they're the same, wonder if we added special logics to it?

Reply:
Sure. We can use the reference in the first place then iterating as needed.